### PR TITLE
Fix Custom Rules Error

### PR DESCRIPTION
**Pull request by PartMan7**
`/challenge [Gen 8] OU @@@ +Reshiram` failed earlier, since the rule wasn't trimmed.

Also going to add the fix for chat only including the first rule.

### CODE REVIEW

**Comment by PartMan7** [comment]
Okay, so the issue is that this.splitTarget disregards all arguments past the second comma, since the split cap returns `[user, format @@@ firstRule]` instead of `[user, format @@@ firstRule, nextRule, extraRuleForNoReason]`. I can try editing it to use all rules, but this.splitTarget is a widely used function - is there anything that changing it might break?

File: sim/dex.ts [hunk 1]
```diff
@@ -581,7 +581,7 @@ export class ModdedDex {
 		if (!customRulesString) return format.id;
 		const ruleTable = this.getRuleTable(format);
 		const customRules = customRulesString.split(',').map(rule => {
-			const ruleSpec = this.validateRule(rule);
+			const ruleSpec = this.validateRule(rule.trim());
```
Comment (Zarel) [inline sim/dex.ts:584]: Maybe this should be inside `validateRule`?

**Comment by Zarel** [comment]
We should probably split by `@@@` before using `splitTarget`.

**Comment by Zarel** [comment]
Also, `splitTarget` shouldn't matter as long as the first `,` is guaranteed to be before the first `@@@`, because `splitTarget` only splits on the first `,` instead of all of them.

**Comment by Zarel** [comment]
I've been wanting to fix this but haven't had time, so thanks for tackling this, by the way!

**Comment by PartMan7** [comment]
Oh, didn't notice the further comments. I'll get on them.

File: sim/dex.ts [hunk 2]
```diff
@@ -953,6 +953,7 @@ export class ModdedDex {
 		switch (rule.charAt(0)) {
 		case '-':
 		case '+':
+			rule = rule.trim();
```
Comment (Zarel) [inline sim/dex.ts:956]:
```suggestion
		rule = rule.trim();
		switch (rule.charAt(0)) {
		case '-':
		case '+':
```

(This is necessary if the rule starts with a space.)

**Comment by Zarel** [comment]
Apparently this entire thing is kind of convoluted. I might need to take a look myself to figure out exactly which part is missing trimming.

**Comment by PartMan7** [comment]
When you say `part`, which are you talking about? I thought the rules were the only things missing trimming. 'o.o

**Comment by Zarel** [comment]
The validation code is pretty complicated, and I would need to look at every level to figure out which step the trimming should be happening in.

**Comment by Zarel** [comment]
It's plausible to me that `validateRule` should simply fail if the input is untrimmed; the code that calls it looks like it would bug out with untrimmed input anyway.

**Comment by KrisXV** [comment]
I would like to bump this because this issue is still affecting `/chall`

**Comment by KrisXV** [comment]
bumping again @PartMan7 @Zarel

**Comment by PartMan7** [comment]
I thought Zarel wanted to suggest some changes; should I mark this as ready?

**Comment by PartMan7** [comment]
@Zarel Hiya?

**Comment by Zarel** [comment]
Oh sorry I'll prioritize this.

**Comment by PartMan7** [comment]
At this point, I'm setting it as Ready for Review since there's been no comments at all.

**Comment by Zarel** [comment]
I'm still working on this, although I sympathize with your impatience. Rest assured this is high priority to me.

**Comment by Zarel** [comment]
Okay, this is definitely a good step. But I'm still confused about the `splitTarget` issue?

**Comment by Zarel** [comment]
I tested `/challenge [Gen 8] OU @@@ +Reshiram, +Zekrom` and it seems to be working?

**Comment by PartMan7** [comment]
When I tried it earlier, the challenge code you used wouldn't allow Zekrom. The first one would work as normal.

**Pull request merged**